addons/website_payment/static/src/snippets/s_donation/000.js

```diff
@@ -24,6 +26,14 @@ publicWidget.registry.DonationSnippet = publicWidget.Widget.extend({
             this._setBubble(this.$rangeSlider);
         }
         await this._displayCurrencies();
+        const customButtonEl = this.el.querySelector("#s_donation_amount_input");
+        if (customButtonEl) {
+            const canvasEl = document.createElement("canvas");
+            const context = canvasEl.getContext("2d");
+            context.font = window.getComputedStyle(customButtonEl).font;
+            const width = context.measureText(customButtonEl.placeholder).width;
+            customButtonEl.style.maxWidth = `${Math.ceil(width) + CUSTOM_BUTTON_EXTRA_WIDTH}px`;
```

Comment:
Doesn't this works?
Less line, no need of creating canvas, using context etc
```js
customButtonEl.setAttribute('size',customButtonEl.getAttribute('placeholder').length);
```
And I am not even sure we want to remove this in the cleanForSave, as it's recomputed at every snippet start, but I'd let you decide if you want to keep it or not

Reply:
Unfortunately, the `size` attribute does not take precedence over the CSS rules involved in the form layout. In Firefox `size` is taken into account, but in Chrome there seems to be some implicit CSS rules that impacts the width which prevent `size` from being taken into account for `input[type="number"]`.
Also, the width obtained through the use of `size` uses a wide character of the font (something like `size * "W"'s width`) - which adds a wide gap after the placeholder string.

Regarding the `cleanForSave`, I believe we want to avoid saving such dynamically obtained parameters.

Reply:
Indeed, according to the spec, `size` does not apply on `input[type="number"]` as its size is handled through the `min` and `max` attributes which defines the length based on the allowed ranged.
I think it's a bit weird, wouldn't hurt to make it work on those input, but it's how it is.
Thanks for the explanation.